src/rpc_connection.c

```diff
@@ -824,13 +826,44 @@ rpc_recv_msg(struct rpc_connection *conn, const void *frame, size_t len,
 	return (0);
 }
 
+static void
+call_abort_locked(struct rpc_call *call)
+{
+	rpc_abort_handler_t abt_fn;
+
+	abt_fn = call->rc_abort_handler;
+	call->rc_abort_handler = NULL;
+	g_mutex_unlock(&call->rc_mtx);
```

Comment:
This function takes the call locked, but leaves it unlocked. It's a confusing practice. Please rework it in a way that matching lock operation can be easily found in the same function. I noticed that both usages of `call_abort_locked()` are quite similar - I think this code can be further de-duplicated to get rid of this problem.

Reply:
There are possible races in different threads for the abort handler. The lock has to be dropped before calling the abort handler. Another thread may execute the abort handler and clean it up as soon as the lock is dropped. Therefore the abort function is saved and nulled out in the call structure before the lock is dropped.